### PR TITLE
📖 Backport: document how to verify control-plane is running

### PR DESCRIPTION
**Pull request by wfernandes**
**What this PR does / why we need it**:
This PR adds a line of documentation that provides a way for users to verify if their control-plane machines are up and running.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Ref #2074

### CODE REVIEW

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -640,6 +640,12 @@ spec:
 {{#/tab }}
 {{#/tabs }}
 
+To verify the control plane is up and running ensure the control plane machine
+`PHASE` is `running`.
+```
+kubectl get machines --selector cluster.x-k8s.io/control-plane=true
+```
```
Comment (detiber) [inline docs/book/src/user/quick-start.md]: Is the Machine Phase of running sufficient here? I would expect the Machine to be running prior to the control plane being available (bootstrapping has not yet completed)

File: docs/book/src/user/quick-start.md [hunk 2]
```diff
@@ -1179,6 +1185,3 @@ spec:
 
 {{#/tab }}
 {{#/tabs }}
-
-<!-- links -->
-[target cluster]: ../reference/glossary.md#target-cluster
```
Comment (wfernandes) [inline docs/book/src/user/quick-start.md]: Decided to remove this link for now to reduce confusion until we have proper Glossary terminology. 
See this issue/comment for more details: https://github.com/kubernetes-sigs/cluster-api/pull/2042#issuecomment-576767956

File: docs/book/src/user/quick-start.md (hunk 1 above)
Comment (wfernandes) [reply docs/book/src/user/quick-start.md]: I believe it is the bootstrap controller that is responsible for setting the Machine state to Running when a ProviderID is set and a Kubernetes Node with same ProviderID is in Ready state. 
_Instead of mentioning the phase we could specify if a ProviderID is available._ 

However, if your comment is regarding that the node may be Ready but the control plane API isn't accepting requests yet then we wouldn't be able to verify the control plane is up and running unless we try and target the cluster with the kubeconfig which is the next step anyways.

Comment (detiber) [reply docs/book/src/user/quick-start.md]: sgtm

File: docs/book/src/user/quick-start.md [hunk 3]
```diff
@@ -640,7 +640,13 @@ spec:
...
 {{#/tabs }}
 
-After the controlplane is up and running, let's retrieve the [target cluster] Kubeconfig:
+To verify the control plane is up, check if the control plane machine
+has a ProviderID.
+```
+kubectl get machines --selector cluster.x-k8s.io/control-plane
+```
+
+After the controlplane is up, we can retrieve the target cluster Kubeconfig:
```
Comment (detiber) [inline docs/book/src/user/quick-start.md]:
```suggestion
After the controlplane is up, we can retrieve the workload cluster Kubeconfig:
```

**Pull request merged**